src/community/label_propagation.c

```diff
@@ -259,7 +260,9 @@ igraph_error_t igraph_community_label_propagation(const igraph_t *graph,
         igraph_vector_int_t *ineis;
         igraph_bool_t was_zero;
 
-        running = 0;
+        if (control_iteration) {
+            running = 0;
+        }
 
         /* Shuffle the node ordering vector */
         IGRAPH_CHECK(igraph_vector_int_shuffle(&node_order));
```

Comment:
You only need to shuffle the nodes when not doing a control iteration (the shuffling is immaterial for checking for dominance).